content/developers/agent_checks.md

```diff
@@ -442,6 +442,52 @@ class HTTPCheck(AgentCheck):
         })
 ```
 
+## Writing checks that run command line programs
+
+It's possible to create a custom check that runs a command line program and captures its output as a custom metric. For example, a check can run the `vgs` command to report information about volume groups. Since the Python code that runs the checks is embedded in the multi-threaded Go runtime, using the Python `subprocess` or `multithreading` modules from the standard library can cause the Agent to crash or processes to get stuck in a zombie state.
+
+To run a subprocess within a check, use the `get_subprocess_output()` function provided in `datadog_checks.utils.subprocess_output`:
+
+```python
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+
+class MyCheck(AgentCheck):
+    def check(self, instance):
+    # [...]
+    out, err, retcode = get_subprocess_output(cmd, self.log, raise_on_empty_output=True)
+```
+
+The command and its arguments are passed to `get_subprocess_output()` in the form of a list, with the command and each of its arguments as a string within the list. For instance, a command that is entered at the command prompt like this:
+
+```
+$ vgs -o vg_free
+
+```
+should be passed to `get_subprocess_output()` like this:
```

Comment:
r/`should be passed`/`must be passed` per [RFC 2119] (https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#rfc-2119)